File: CPPLib/src/TCPWrapper.cpp
#include "comms/GenericNetWrapper.hpp"
#include "comms/TCPWrapper.hpp"

int newTCPServer(const std::string& Host, int Port) {
    int server_fd = socket(AF_INET, SOCK_STREAM, 0);

    int opt = 1;
    setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt));
    setsockopt(server_fd, SOL_SOCKET, SO_REUSEPORT, &opt, sizeof(opt));

    sockaddr_in serv{};
    serv.sin_family = AF_INET;
    serv.sin_port = htons(Port);
    inet_pton(AF_INET, Host.c_str(), &serv.sin_addr);

    bind(server_fd, (sockaddr*)&serv, sizeof(serv));
    listen(server_fd, 1);

    return server_fd;
}


TCPClient::TCPClient(const std::string& Host, int Port) {
    sockfd = socket(AF_INET, SOCK_STREAM, 0);

    sockaddr_in serv{};
    serv.sin_family = AF_INET;
    serv.sin_port = htons(Port);
    inet_pton(AF_INET, Host.c_str(), &serv.sin_addr);

    connect(sockfd, (sockaddr*)&serv, sizeof(serv));

    //Set up the statistics
    double now = time(nullptr);
    this->info.InitTime = now;
    this->info.LastPacket = now;
    this->info.IP = Host;
    this->info.Port = Port;
}

TCPClient::~TCPClient() {
    this->close(sockfd);
}

File: PythonLib/DEFCOM/_TCPWrapper.py
#!/usr/bin/env python3
import socket
from ._GenericNetWrapper import genericNetWrapper

class clientTCPCon(genericNetWrapper):
  def __init__(self,Host,Port):
    conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    conn.connect((Host, Port))

    super().__init__(conn)



def newTCPServer(Host,Port):
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
    s.bind((Host, Port))
    s.listen(1)
    return s
    
    
class serverTCPCon(genericNetWrapper):
    def __init__(self,Server):
      conn, addr = Server.accept()

      super().__init__(conn,addr)
      
      
